Plugin/ClearQuote.php

```diff
@@ -25,14 +28,69 @@
  */
 class ClearQuote
 {
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /**
+     * @var \Magento\Quote\Model\Quote|null
+     */
+    private $quote_to_restore;
+
+    /**
+     * @param CartHelper $cartHelper
+     */
+    public function __construct(
+        CartHelper $cartHelper
+    )
+    {
+        $this->cartHelper = $cartHelper;
+    }
+
+    /**
+     * @param CheckoutSession $subject
+     * @return CheckoutSession
+     * @throws \Exception
+     */
+    public function beforeClearQuote(CheckoutSession $subject)
+    {
+        // We don't want to clear quote
+        // in Product page checkout (PPC) flow
+        $this->quote_to_restore = null;
+        $current_quote_id = $subject->getQuote()->getId();
+        $order_quote_id = $subject->getLastSuccessQuoteId();
+        if (!$current_quote_id || !$order_quote_id && $current_quote_id == $order_quote_id) {
+            // In PPC checkout quote should be different then quote tied to order just created
+            return $subject;
+        }
+
+        // Although check above is enough, double check that we are in Bolt PPC process
+        $quote = $this->cartHelper->getQuoteById($order_quote_id);
+        if (!$quote || $quote->getBoltParentQuoteId() != $order_quote_id) {
+            // BoltParentQuoteId should be set (sign of Bolt)
+            // and should be the same as quoteID (sign of PPC)
+            return $subject;
+        }
+
+        $this->quote_to_restore = $subject->getQuote();
+
+        return $subject;
```

Comment:
Thank you Daniel! It seems by documentation that we need to return "null" because we don't want to change arguments.
>  If the method does not change the argument for the observed method, it should return a null value.
What do you think?
https://devdocs.magento.com/guides/v2.3/extension-dev-guide/plugins.html

Reply:
I agree.

Reply:
Done